# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/MessageEventRunnable.cpp
# Commit: b1ab6c750d04
# Full Hash: b1ab6c750d04299011c1446b1752c2ff98a92b33
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-06-07 03:30:33
# Regressor Bug: 1800659
# File Overlap Count: 1
# Description:
#   Bug 1800659 - P3 Remove WorkerPrivate::ClearMainEventQueue() r=asuth,smaug
#   
#   Pending->Canceling->Killing (WorkerNeverRan)
#   
#   Need to clear WorkerPrivate::mPreStartRunnables
# ==============================================================================

diff -r 32515f9b41c7 -r b1ab6c750d04 dom/workers/MessageEventRunnable.cpp
--- a/dom/workers/MessageEventRunnable.cpp	Tue Jun 06 06:36:50 2023 +0000
+++ b/dom/workers/MessageEventRunnable.cpp	Tue Jun 06 06:36:50 2023 +0000
@@ -135,6 +135,13 @@
   }
 
   MOZ_ASSERT(aWorkerPrivate == GetWorkerPrivateFromContext(aCx));
+  MOZ_ASSERT(aWorkerPrivate->GlobalScope());
+
+  // If the worker start shutting down, don't dispatch the message event.
+  if (NS_FAILED(
+          aWorkerPrivate->GlobalScope()->CheckCurrentGlobalCorrectness())) {
+    return true;
+  }
 
   return DispatchDOMEvent(aCx, aWorkerPrivate, aWorkerPrivate->GlobalScope(),
                           false);